hanadb_exporter/exporters/prometheus_exporter.py

```diff
@@ -62,20 +67,23 @@ def collect(self):
         execute db queries defined by metrics_config/api file, and store them in
         a prometheus metric_object, which will be served over http for scraping e.g gauge, etc.
         """
-
+        logger = logging.getLogger(__name__)
         for query in self._metrics_config.queries:
             if not query.enabled:
                 self._logger.info('Query %s is disabled', query.query)
             elif not utils.check_hana_range(self._hana_version, query.hana_version_range):
                 self._logger.info('Query %s out of the provided hana version range: %s',
                                   query.query, query.hana_version_range)
             else:
-                # TODO: manage query error in an exception
-                query_result = self._hdb_connector.query(query.query)
-                formatted_query_result = utils.format_query_result(query_result)
-                for metric in query.metrics:
-                    if metric.type == "gauge":
-                        metric_obj = self._manage_gauge(metric, formatted_query_result)
-                        yield metric_obj
-                    else:
-                        raise NotImplementedError('{} type not implemented'.format(metric.type))
+                try:
+                    query_result = self._hdb_connector.query(query.query)
+                    formatted_query_result = utils.format_query_result(query_result)
+                    for metric in query.metrics:
+                        if metric.type == "gauge":
+                            metric_obj = self._manage_gauge(metric, formatted_query_result)
+                            yield metric_obj
+                        else:
+                            raise NotImplementedError('{} type not implemented'.format(metric.type))
+                except hdb_connector.connectors.base_connector.QueryError as err:
+                    logger.error('Failure in query: %s, skipping...', query.query)
```

Comment:
I think we can improve the error msg

`error by executing query`

Reply:
the current error msg doesn't show the action in log

Reply:
@MalloZup which action are you referring ? It's showing which query is failing and it's showing that the query is skipped.

Reply:
`logger.error('Failure in query: %s, skipping...', query.query)`

with
`logger.error('executing SQL query failed: %s, skipping...', query.query)`